blog/tanstack-router-loaders/index.md

```diff
@@ -0,0 +1,630 @@
+
+If you're wondering why we'd even want to use `pendingMs` to delay a loading screen, it's for subsequent navigations. Rather than _immediately_ transition from the current page to a new page's loading component, this setting lets us stay on the current page for a moment, in the hopes that the new page will be ready quickly enough that we don't have to show any pending component at all. Of course, on the initial load, when the web app first starts up, these pendingComponents do show immediately, as you'd expect.
+
+Let's run our tasks page
+
+![loaders running in parallel](/tanstack-router-loaders/img-0-tasks-page.jpg)
+
+It's ugly and frankly useless, but it works. Now let's take a closer look.
+
+### Loaders running in parallel
+
+If we peak in our dev tools, we should see something like this
+
+![loaders running in parallel](/tanstack-router-loaders/img-1-loaders-parallel.jpg)
+
+As we can see, these requests started a mere millisecond apart from each other, since the loaders are running in parallel (each request takes at least 750ms, due to the artificial delays in the api endpoints).
+
+#### Different routes using the same data
+
+If we look at the loader for the `app/tasks/$taskId` route, and the loader to the `app/tasks/$taskId/edit` route, we see the same fetch call
+
+```ts
+const task = await fetchJson<Task>(`api/tasks/${taskId}`);
+```
+
+This makes sense since we need to load the actual task in order to display it, or in order to display it in a form for the user to make changes. Unfortunately though, if you click the edit button for any task, then go back to the tasks list (without saving anything), then click the edit button for the same task, you should notice the same exact data being requested. This makes sense. Both loaders happen to make the same fetch() call, and there's nothing in our client to cache the call. This is probably fine 99% of the time, but this is one of the many things react-query will improve for us, in a bit.
+
+## Updating data
+
+If you click the edit button for any task, you should be brought to a page with an extremely basic form that will let you edit the task's name. Once we click save, we want to navigate back to the tasks list, but most importantly, we need to tell Router that we've changed some data, and that it will need to invalidate some cached entries, and re-fetch when we go back to those routes.
+
+This is where Router's built-in capabilities start to get stretched, and where we might start to want react-query (discussed below). Router will absolutely let you invalidate routes, to force re-fetches. But eh api is fairly simple, and fine-grained. We basically have to describe each route we want invalidated (or removed). Let's take a look:
```

Comment:
> But eh api

=> the?

Reply:
> We basically have to describe each route we want invalidated (or removed).

you don't HAVE to. leaving out the filter will invalidate everything

Reply:
fair - updated